### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by fontikar**
Addresses Reviewer 1's suggestion on how to cite the version of the package as well as the taxonomic resources.

Changes made

- An article on the website walking a reader through how to get the package version and version of the taxonomic resource
- I have added a citation file so folks can cite the package and eventually the research article when in press
- Exported default_version() so users can see what version of taxonomic resources is being used in alignment and update.
- Updated website yml so new article will show
- While I was updating website, I fixed some bugs about missing images in data_provider article and the name to call the 'updating-taxon-names" vignette
- Have bumped version number and updated NEWS.md accordingly

### CODE REVIEW

File: _pkgdown.yml [hunk 1]
```diff
@@ -18,19 +18,22 @@ navbar:
       menu:
       - text: "Data providers"
       - text: APC and APNI
-        href: articles/data-providers.html
+        href: articles/articles/data-providers.html
```
Comment (dfalster) [inline _pkgdown.yml:21]: surely wrong?

File: inst/CITATION [hunk 2]
```diff
@@ -0,0 +1,14 @@
+bibentry(
+  bibtype  = "Manual",
```
Comment (dfalster) [inline inst/CITATION:2]: Shall we change this to the preprint? https://www.biorxiv.org/content/10.1101/2024.02.02.578715v1

File: vignettes/articles/reproducibility.Rmd [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+```{r}
+packageVersion("APCalign")
+```
+
+#### Taxonomic Resources
+
+`APCalign` allows users to load static downloads of taxonomic resources the APC and APNI or the latest version from the National Species List website. This functionality is specified using the `stable_or_current_data` argument of `load_taxonomic_resources()`. 
+
+We recommend to always use `stable_or_current_data = "stable"` if you want your taxonomic alignment and update to be reproducible. The preset value is `stable_or_current_data = "stable"`. These static downloads are version controlled and stored in our repository as [releases](https://github.com/traitecoevo/APCalign/releases).
```
Comment (dfalster) [inline vignettes/articles/reproducibility.Rmd:41]:
```suggestion
If you want your taxonomic alignment and update to be reproducible, we recommend to always use `stable_or_current_data = "stable"`. The default value is `stable_or_current_data = "stable"`. These static downloads are version controlled and stored in our repository as [releases](https://github.com/traitecoevo/APCalign/releases). 
```

File: vignettes/articles/reproducibility.Rmd [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+Then copying and pasting the output into `load_taxonomic_resources()` directly. This way makes the version of taxonomic resources more explicit in your code.
+
+```{r}
+resources <- load_taxonomic_resources(
+  stable_or_current_data = "stable",
+  version = "0.0.4.9000"
+)
+```
```
Comment (dfalster) [inline vignettes/articles/reproducibility.Rmd:69]: This is good, but can we add something here that shows how to the pass the resources into the successive functions?

Comment (dfalster) [reply vignettes/articles/reproducibility.Rmd:69]: Also, show the capability to save the results to file and reuse those

**Review by dfalster — CHANGES_REQUESTED** [review]
Nice, some suggested additions!

File: _pkgdown.yml (hunk 1 above)
Comment (fontikar) [reply _pkgdown.yml:21]: Nope! Trust ;) I thought I was going crazy. Removing the second articles breaks the image file path.

File: inst/CITATION (hunk 2 above)
Comment (fontikar) [reply inst/CITATION:2]: Great idea! Will update the authors according too!

File: vignettes/articles/reproducibility.Rmd (hunk 4 above)
Comment (fontikar) [reply vignettes/articles/reproducibility.Rmd:69]: > Also, show the capability to save the results to file and reuse those

Although our functions have the option to save output of alignments and updates. I don't actually think it currently works. 

```
aligned_updated_taxa <- create_taxonomic_update_lookup(gbif_lite$species, resources = resources, output = "ignore/")
Checking alignments of 121 taxa

  -> 0 names already matched; 0 names checked but without a match; 121 taxa yet to be checked
Error in open.connection(3L, "wb") : cannot open the connection
In addition: Warning messages:
1: In file(path) :
  'raw = FALSE' but '/Users/fontikar/Dropbox/0 - Job/APCalign/ignore' is not a regular file
2: In open.connection(3L, "wb") :
  cannot open file '/Users/fontikar/Dropbox/0 - Job/APCalign/ignore': Is a directory
  ```
  
  This was on [the backlog ](https://github.com/traitecoevo/APCalign/issues/140) but got deprioritised. 
  
  Should we remove this functionality for now? Thoughts @dfalster

Comment (fontikar) [reply vignettes/articles/reproducibility.Rmd:71]: Addressed in https://github.com/traitecoevo/APCalign/issues/194 and https://github.com/traitecoevo/APCalign/pull/198

**Pull request merged**